### PR TITLE
Don't render 'Contact details' panel if there's no 'P'-type contact (PTCD-140)

### PR DESCRIPTION
**Pull request by gunndabad**

### CODE REVIEW

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cs [hunk 1]
```diff
@@ -67,21 +57,31 @@ public async Task<ViewModel> Handle(Query request, CancellationToken cancellatio
...
+                    GotContact = false
+                };
+            }
+
+            var contactName = contact.ContactPersonalDetails?.PersonGivenName != null && contact.ContactPersonalDetails?.PersonFamilyName != null ?
                 string.Join(" ", contact.ContactPersonalDetails.PersonGivenName) + " " +
                     contact.ContactPersonalDetails.PersonFamilyName :
                 null;
 
             return new ViewModel()
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cs:74]: Consider `ViewModel.FromProvider(Provider provider)...` method to remove some of this logic from the controller. (suggestion, not request)

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml [hunk 2]
```diff
@@ -1,53 +1,56 @@
...
-        </div>
-    }
-    @if (!string.IsNullOrEmpty(Model.Website))
-    {
-        <div class="govuk-!-margin-bottom-4" id="pttcd-new-apprenticeship-provider__provider-detail-info-panel__provider-website">
-            <h3 class="govuk-heading-m m20 govuk-!-margin-0" >Web</h3>
-            <div class="govuk-!-font-size-16">
-                <a href="@Model.Website" class="govuk-link" rel="noreferrer" target="_blank">@Model.Website</a>
+        }
+        @if (!string.IsNullOrEmpty(Model.Website))
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml:19]: Is this if still required?

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml [hunk 3]
```diff
@@ -1,53 +1,56 @@
...
-        </div>
-    }
-    @if (!string.IsNullOrEmpty(Model.ContactName))
-    {
-        <div class="govuk-!-margin-bottom-4" id="pttcd-new-apprenticeship-provider__provider-detail-info-panel__provider-contact-name">
-            <h3 class="govuk-heading-m m20 govuk-!-margin-0" >Contact name</h3>
-            <div class="govuk-!-font-size-16">
-                @Model.ContactName
+        }
+        @if (!string.IsNullOrEmpty(Model.ContactName))
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml:28]: Is this if still required?

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml [hunk 4]
```diff
@@ -1,53 +1,56 @@
...
-        </div>
-    }
-    @if (!string.IsNullOrEmpty(Model.Email))
-    {
-        <div class="govuk-!-margin-bottom-4" id="pttcd-new-apprenticeship-provider__provider-detail-info-panel__provider-email">
-            <h3 class="govuk-heading-m m20 govuk-!-margin-0">Email</h3>
-            <div class="govuk-!-font-size-16">
-                <a href="mailto:@Model.Email" class="govuk-link">@Model.Email</a>
+        }
+        @if (!string.IsNullOrEmpty(Model.Email))
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml:37]: Is this if still required?

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml [hunk 5]
```diff
@@ -1,53 +1,56 @@
...
-        </div>
-    }
-    @if (!string.IsNullOrEmpty(Model.Telephone))
-    {
-        <div class="govuk-!-margin-bottom-4" id="pttcd-new-apprenticeship-provider__provider-detail-info-panel__provider-telephone">
-            <h3 class="govuk-heading-m m20 govuk-!-margin-0" >Telephone</h3>
-            <div class="govuk-!-font-size-16">
-                @Model.Telephone
+        }
+        @if (!string.IsNullOrEmpty(Model.Telephone))
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml:46]: Is this if still required?

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cs (hunk 1 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cs:74]: The current V2 approach is having dumb VM classes (POCOs) then having the mapping logic in these handler types.

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml (hunk 2 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml:19]: Yep - you can have a non-null contact where this field is empty

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml (hunk 3 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml:28]: As above - you can have a non-null contact where this field is empty

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml (hunk 5 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml:46]: As above - you can have a non-null contact where this field is empty

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml (hunk 4 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cshtml:37]: As above - you can have a non-null contact where this field is empty

File: src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cs [hunk 6]
```diff
@@ -3,16 +3,15 @@
...
 using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Models;
 using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
 using Dfc.CourseDirectory.Core.DataStore.Sql;
-using Dfc.CourseDirectory.Core.DataStore.Sql.Queries;
-using Dfc.CourseDirectory.Core.Models;
+using Dfc.CourseDirectory.WebV2.Behaviors.Errors;
 using Dfc.CourseDirectory.WebV2.Security;
 using MediatR;
-using Dfc.CourseDirectory.Core;
+using Microsoft.SqlServer.TransactSql.ScriptDom;
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ProviderDetailInfoPanel.cs:14]: Unnecessary using?

**Pull request merged**